src/controllers/file.ts

```diff
@@ -0,0 +1,108 @@
+import { Application, Request, Response } from 'express'
+import * as path from 'path'
+
+var fs = require('fs');
+
+import {availableDrives, UsbDeviceInfo} from './device'
+
+const _writeDirectory = async (deviceId: number, directoryName: string)
+: Promise<{success: boolean, message: string}> => {
+    return new Promise(async resolve => {
+
+      let device = availableDrives[deviceId]
+      console.log(device)
+
+      if (device) {
+        let localPath = path.join(device.mountpoints[0], directoryName)
+
+        let result = fs.existsSync(localPath) || fs.mkdirSync(localPath);
+      
+        resolve({ 
+          success: true, 
+          message: result
+        })
+      } else {
+        resolve({ 
+          success: false,
+          message: "no candidates found" 
+        })
+      }
+    })
+  }
+
+const _writeFile = async (deviceId: number, fileName: string, content: string)
```

Comment:
No need for `async` here or below in the `Promise` constructor... you're not awaiting anything.